scipy/sparse/linalg/_isolve/tests/test_iterative.py

```diff
@@ -427,24 +420,28 @@ def test_zero_rhs(solver):
         with suppress_warnings() as sup:
             sup.filter(DeprecationWarning, ".*called without specifying.*")
 
-            x, info = solver(A, b, tol=tol)
+            if solver in [cg, cgs, bicg, bicgstab, gmres]:
+                tol_kwarg = {'rtol': tol}
+            else:
+                tol_kwarg = {'tol': tol}
+            x, info = solver(A, b, **tol_kwarg)
```

Comment:
Same here. This is already in `suppress_warnings()` block, so we just need to add one more filter on top